docs/connectivity/Agent.mdx

```diff
@@ -681,6 +681,51 @@ To update a connection, select the desired connection in the Connections tab to
+
+- Core Version Update: The core version corresponds to the underlying agent code developed by Ganymede. These changes are new features or patches that can be applied to all agents across all environment but each agent manages which core version it is pinned to.
+- Agent Configuration Update: Agent configuration is updated through the agent page or the agent's processor notebook. This is a configuration that all connections for that agent adhere to. For example, if an agent that watcher for files then runs a flow is updated to run a new flow when the params are met, all compatible connections will begin to run that flow after they complete their update. Every save to an agent notebook is considered a new version that will be retrieved and loaded after the connections next status ping.
+- Connection Configuration Update: Updates made to the connection's configuration on its page will only update the running connection that it corresponds to. Other connections for that agent will not observe any change is made. This includes connection aspects such as variables and labels.
+
+There are two types of remote updates that occur based on the type of update - full or runtime update.
+
+#### Full Update
+**Compatibility**
+- Agents _installed_ with core v4.8.0 or later
+- The core version of the installed connection must align with the major version of the core version of the agent Ganymede’s UI. Otherwise there is a breaking change that requires reinstallation.
+- Auto-update is enabled for the agent
+
+**Cause**
+A full update is only triggered for a "Core Version Update." The remote update is not kicked off without user intervention. In the Ganymede UI, an agent's core version is listed on the right side of the page in a section called "Core Agent Infrastructure." If the agent is not up to date to the latest available core version, an "Upgrade" button will appear. The text will inform users what version is available and what version the agent is currently pinned to. 
+
+<div class="text--center">
+  <img
+    alt="Agent build history"
+    src="https://ganymede-bio.mo.cloudinary.net/agent/AgentUpgrade.png"
+    width="800"
+  />
+</div>
+
+Once a user clicks to Upgrade an agent, all connections will be told of the new version on their next status ping. 
+
+**Update Process**
+A full update requires reinstalling the executable, but Ganymede now takes care of this installation seamlessly. When a connection receives a command to do a full update, it will continue to operate normally but also attempt to start the new version in parallel after verifying the validity of the command and the executable. Once this new version registers with Ganymede as operational, the old connection will stop. The new executable will also cleanup the artifacts of the old executable and its dependencies.
```

Comment:
```suggestion
A full update requires reinstalling the executable, but Ganymede now takes care of this installation seamlessly. When a connection receives a command to do a full update, it will continue to operate normally but also attempt to start the new version in parallel after verifying the validity of the command and the executable. Once this new version registers with Ganymede as operational, the old connection will stop. The new executable will also cleanup the artifacts of the old executable and its dependencies.

There may be a disruption to any ongoing processing on the agent when an update is in progress so it is safest to request it when there is no activity on the instruments where the agents are running.
```